### PR TITLE
Wire Up APIs On FE

### PR DESCRIPTION
**Pull request by nickerick**
## Issue
Frontend has all hardcoded data currently

## Solution
Wired up Course API and wired up the following screens:
- Course Marketplace
  -  Courses are now generated from the `getAllCourses()` endpoint
- My Courses
  - Courses are now generated for a user's enrolled courses using the `getEnrolledCourses(userId)` endpoint
- Course Viewer :star:
  - Courses and lessons are generated from `getCourse(courseId)` endpoint 
  - Uploaded dummy videos that match up with the dummy data in our local dev environment
 
Cleaned up various screens to dynamically generate data.

### CODE REVIEW

File: client/src/services/CourseService.js
Comment (sjdefran1) [inline client/src/services/CourseService.js:1]: looks good and easy to replicate! good work

File: client/src/pages/mycourses/MyCourses.jsx [hunk 1]
```diff
@@ -1,23 +1,45 @@
...
+          {courses.map(c => (
+            <MyCourseCard
+              key={c.enrolledCourse.id}
+              courseTitle={c.enrolledCourse.title}
+              imageUrl=""
+              instructorName={c.enrolledCourse.description}
+              progress={c.progress}
+              handleClick={() => navigate(`/courseviewer/${c.id}`)}
+            />
+          ))}
```
Comment (sjdefran1) [inline client/src/pages/mycourses/MyCourses.jsx:40]: love me some mapping :)

**Review by sjdefran1 — APPROVED** [review]
Looks good, good stuff

File: client/src/pages/mycourses/MyCourses.jsx (hunk 1 above)
Comment (nickerick) [reply client/src/pages/mycourses/MyCourses.jsx:40]: Absolute favorite part about coding. Functional programming FTW. 

I used to not be a huge fan of it and things like [this](https://github.com/nickerick/skill-magnet/blob/288668759b96ecaf8c01f130985f6133b32a8818/server/src/main/java/com/skillmagnet/Lesson/LessonController.java#L69) were the most I approved of. But at my current internship they have some pretty disgusting mappings that go on where it'll be like 15 methods chained together. I've been stockholmed syndromed into loving it now. Also where Optionals are super useful and make much more sense than how we're using them lol

If you go down to the 'Less Boilerplate' section of this [article](https://belief-driven-design.com/functional-programming-with-java-streams-190eda591a5/) you can be gateway drugged into the functional programming cult.

Comment (sjdefran1) [reply client/src/pages/mycourses/MyCourses.jsx:40]: > I've been stockholmed syndromed into loving it now

lmao

> Also where Optionals are super useful and make much more sense than how we're using them lol

I definitely see the potential.

```java
List<String> result = 
    albums.stream()
          .filter(album -> album.getYear == 1999)
          .filter(album -> album.getGenre() == Genre.ALTERNATIVE)
          .limit(5)
          .map(Album::getName)
          .sorted()
          .collect(Collectors.toList());
```
This is beautiful. Doesn't even look like java to me, if you know what I mean lol

Comment (nickerick) [reply client/src/pages/mycourses/MyCourses.jsx:40]: Life is good when all your logic looks like this. It's so alien at first but once you get used to it you realize how much faster it is to write code and also figure out what something is doing

**Pull request merged**